### PR TITLE
Fixes #5991: Adds Convert Symlinks Option

### LINKED ISSUE #5991: Symlinks to directories outside project break archive dump.
**Describe the bug**
When doing `archive-dump` for code, if a symlink in the code leads to a directory outside the project code being archived, an error will be thrown.

**To Reproduce**
1. Create a symlink inside a project whose linked content is outside the project
2. Run `drush archive:dump --code`

**Expected behavior**
An archive will be generated successfully.

**Actual behavior**
```
In ArchiveDumpCommands.php line 213:

  [UnexpectedValueException]
  Iterator RecursiveIteratorIterator returned a path "{{symlink-destination}}" that is not in
  the base directory "{{tmp-directory}}"
```

**Workaround**
Remove the symlink

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 10.x/11.x/12.x 
| Drupal version? | 10.x/9.x/8.x/7.x
| PHP version     | 8.x/7.x
| OS?             | Mac/Linux

**Additional information**
This has only been verified in some configurations, but likely exists in most or all of them.

### PR DESCRIPTION
**Pull request by greg-1-anderson**
Ryan Wagner is no longer working at Pantheon, having moved on to Apple's Drupal group (congratulations on the new job); I will therefore be picking this up and finishing it.

### CODE REVIEW

**Comment by greg-1-anderson** [comment]
This should take care of the previous review comments.

File: src/Commands/core/ArchiveDumpCommands.php [hunk 1]
```diff
@@ -237,6 +242,65 @@ private function createManifestFile(array $options): void
         );
     }
 
+    /**
+     * Converts symlinks to the linked files/folders for an archive.
+     *
+     * @param bool $convert_symlinks
+     *  Whether to convert all symlinks.
+     *
+     * @return void
```
Comment (weitzman) [inline src/Commands/core/ArchiveDumpCommands.php]: Lets omit. Our PHPCS is configured to ignore missing Doxygen.
```suggestion
```

File: src/Commands/core/ArchiveDumpCommands.php [hunk 2]
```diff
@@ -237,6 +242,65 @@ private function createManifestFile(array $options): void
...
+        $this->logger()->info(dt('Converting symlinks...'));
+
+        $iterator = new RecursiveIteratorIterator(
+            new RecursiveDirectoryIterator($this->archiveDir),
+            RecursiveIteratorIterator::SELF_FIRST
+        );
+
+        foreach ($iterator as $file) {
+            if (
+                $file->isLink() && ($convert_symlinks === true || strpos(
```
Comment (weitzman) [inline src/Commands/core/ArchiveDumpCommands.php]:
```suggestion
                $file->isLink() && ($convert_symlinks || strpos(
```

File: tests/functional/ArchiveSymlinkTest.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    public function tearDown(): void
+    {
+        unlink($this->linktarget);
+        unlink($this->linkdestination);
+    }
+
+    public function testArchiveDumpSymlinkReplaceCommand(): void
+    {
+        // The symliknks written in setup would cause the PharData class to
```
Comment (weitzman) [inline tests/functional/ArchiveSymlinkTest.php]:
```suggestion
        // The symlinks written in setup would cause the PharData class to
```

**Pull request merged**